Core/src/Detector/DetectorVolumeBuilder.cpp

```diff
@@ -0,0 +1,79 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Detector/DetectorVolumeBuilder.hpp"
+
+#include "Acts/Detector/DetectorVolume.hpp"
+
+#include <stdexcept>
+
+Acts::Experimental::DetectorVolumeBuilder::DetectorVolumeBuilder(
+    const Acts::Experimental::DetectorVolumeBuilder::Config& cfg,
+    std::unique_ptr<const Acts::Logger> logger)
+    : IDetectorComponentBuilder(), m_cfg(cfg), m_logger(std::move(logger)) {
+  if (m_cfg.externalsBuilder == nullptr) {
+    throw std::invalid_argument(
+        "DetectorVolumeBuilder: no external structure builder defined.");
+  }
+}
+
+Acts::Experimental::DetectorComponent
+Acts::Experimental::DetectorVolumeBuilder::construct(
```

Comment:
seeing the return type now maybe it a better name for this factory would be `DetectorComponentBuilder`?

Reply:
The point is that I want to make it clear that this builds a single volume and returns it as a component for successive building, the `ContainerBuilder` will also return a `DetectorComponent`.